docs/style.md

```diff
@@ -539,6 +539,245 @@ import { unstable_Pagination as Pagination } from 'carbon-components-react';
+      }
+    );
+
+    // Optional ref tests
+    // A component that accepts a ref falls in one of three scenarios:
+    // 1. A class component
+    // 2. A component that uses React.forwardRef and placed
+    //    it on an HTML element
+    // 3. A component that uses React.forwardRef and uses
+    //    useImperativeHandle to decorate the ref (this is
+    //    uncommon but can come up)
+  });
+});
+```
+
+##### `ComponentName-test.a11y.js`
+
+- Use `accessibility-checker` and `axe`
+- Optionally configure common props to ensure component variants do not contain
+  accessibility errors.
+
+```js
+describe('ComponentName - AVT1', () => {
+  it('should have no aXe violations', async () => {
+    render(<ComponentName />);
+    await expect(
+      screen.getByLabelText('or a similar query')
+    ).toHaveNoAxeViolations();
+  });
+
+  it('should have no AC violations', async () => {
+    render(<ComponentName />);
+    await expect(
+      screen.getByLabelText('or a similar query')
+    ).toHaveNoACViolations('ComponentName');
+  });
+});
+```
+
+##### `ComponentName-test.e2e.js`
+
+- These tests are ran with Cypress, which is a bit different in syntax and
+  mental model than Jest/JSDOM tests. It's recommended to become familiar with
+  the core concepts by reading the
+  [Cypress introduction documentation](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress).
+- Trigger Events with ``via`cy.real\*();`
+- Keyboard navigation/interaction testing is best done here, in a simulated
+  browser environment, not JSDOM.
+- Take snapshots with Percy
+  - Percy snapshots should be used very sparingly (1-2 per component)
+  - Percy snapshots should always be taken _after_ an assertion that a
+    element/componen should be visible. This ensures the DOM has settled and the
+    element has fully loaded.
+
+```js
+import 'carbon-components/scss/components/accordion/_accordion.scss';
+
+import React from 'react';
+import { mount } from '@cypress/react';
+
+describe('ComponentName - e2e', () => {
+  beforeEach(() => {
+    mount(<ComponentName />);
+  });
+
+  it('should render', () => {
+    cy.findByText(/ComponentText/).should('be.visible');
+
+    cy.findByText(/ComponentText not visible until interaction/).should(
+      'not.be.visible'
+    );
+
+    cy.percySnapshot();
+  });
+
+  describe('keyboard navigation', () => {
+    it('should tab between items and open on space', () => {
+      cy.findByRole('button', { name: /Section 1/i }).focus();
+
+      cy.realPress('Tab');
+      cy.findByRole('button', { name: /Section 2/i }).should('be.focused');
+
+      // ... continued interaction testing a full keyboard navigation flow
+    });
+  });
+});
+```
+
+##### `ComponentName-test.server.js`
+
+```js
+/**
+ * @jest-environment node
+ */
+
+import { ComponentName } from '../ComponentName';
+
+describe('ComponentName - SSR', () => {
+  it('should import ComponentName in node/server env', () => {
```

Comment:
@joshblack Yep, if you'd prefer we can have one file that imports/covers the entire package. Though do you think it would be helpful to stub out these simple SSR test files for each component so we'd have them for future expansion of server side testing? Otherwise it would be a one-off file that not all components would have in their folder.